README.md
## WORK IN PROGRESS
# xxx
xxx

[![Latest Version on Packagist][ico-version]][link-packagist]
[![Software License][ico-license]](LICENSE.md)
[![Build Status][ico-travis]][link-travis]
[![Coverage Status][ico-scrutinizer]][link-scrutinizer]
[![Quality Score][ico-code-quality]][link-code-quality]
[![Total Downloads][ico-downloads]][link-downloads]
[![SensioLabsInsight][ico-sensiolab]][link-sensiolab]

Table of Contents
=================

  * [xxx](#xxx)
  * [Table of Contents](#table-of-contents)
  * [Prerequisites](#prerequisites)
  * [Install](#install)
  * [Usage](#usage)
  * [Example](#example)
  * [Screenshots](#screenshots)
  * [Change Log](#change-log)
  * [Testing](#testing)
  * [Contributing](#contributing)
  * [Security](#security)
  * [API Documentation](#api-documentation)
  * [Credits](#credits)
  * [About Padosoft](#about-padosoft)
  * [License](#license)

# Prerequisites

# Install

This package can be installed through Composer.

``` bash
composer require padosoft/xxx
```
You must install this service provider.

``` php
// config/app.php
'provider' => [
    ...
    Padosoft\Xxx\XxxServiceProvider::class,
    ...
];
```

You can publish the config file of this package with this command:
``` bash
php artisan vendor:publish --provider="Padosoft\Xxx\XxxServiceProvider"
```
The following config file will be published in `config/xxx.php`
``` php
[

]
```

# Usage

## Example

# Screenshots

# Change Log
Please see [CHANGELOG](CHANGELOG.md) for more information what has changed recently.

# Testing

# Contributing

Please see [CONTRIBUTING](CONTRIBUTING.md) and [CONDUCT](CONDUCT.md) for details.

# Security

If you discover any security related issues, please email  instead of using the issue tracker.

# API Documentation

Please see API documentation at http://b2msrl.github.io/xxx

# Credits

- [Padosoft](https://github.com/padosoft)
- [All Contributors](../../contributors)

# About Padosoft
Padosoft is a software house based in Florence, Italy. Specialized in E-commerce and web sites.

# License

The MIT License (MIT). Please see [License File](LICENSE.md) for more information.


[ico-version]: https://img.shields.io/packagist/v/padosoft/xxx.svg?style=flat-square
[ico-license]: https://img.shields.io/badge/license-MIT-brightgreen.svg?style=flat-square
[ico-travis]: https://img.shields.io/travis/padosoft/xxx/master.svg?style=flat-square
[ico-scrutinizer]: https://img.shields.io/scrutinizer/coverage/g/padosoft/xxx.svg?style=flat-square
[ico-code-quality]: https://img.shields.io/scrutinizer/g/padosoft/xxx.svg?style=flat-square
[ico-downloads]: https://img.shields.io/packagist/dt/padosoft/xxx.svg?style=flat-square
[ico-sensiolab]: https://insight.sensiolabs.com/projects/@@@sensiolab/small.png

[link-packagist]: https://packagist.org/packages/padosoft/xxx
[link-travis]: https://travis-ci.org/padosoft/xxx
[link-scrutinizer]: https://scrutinizer-ci.com/g/padosoft/xxx/code-structure
[link-code-quality]: https://scrutinizer-ci.com/g/padosoft/xxx
[link-downloads]: https://packagist.org/packages/padosoft/xxx
// TODO : @@@sensiolabs
[link-sensiolab]: https://insight.sensiolabs.com/projects/@@@sensiolabs


CHANGELOG.md
# Changelog

All Notable changes to `:package_name` will be documented in this file

## alpha - 09.09.16

### Added
- Initial commit


tests/config/sedCommand.sh
#!/bin/bash
sed -i 's/"psr-4": {/"psr-4": { "Padosoft\\\\Xxx\\\\Test\\\\": ".\/vendor\/padosoft\/xxx\/tests\/",/g' ./composer.json

tests/config/sedCommandProvider.sh
#!/bin/bash
sed -i "s/'providers' => \[/'providers' => \[ Padosoft\\\\Xxx\\\\XxxServiceProvider::class,/g" ./config/app.php

cache/master/twig/08/08b20e39e69ec89e3e5d30c897772345de2ba6bd30cc8f1830108fc070e8569d.php
<?php

/* search.twig */
class __TwigTemplate_6b291bcb4c4ea8527a1d1434d86d2c5c24e254571ad48dfa90c2b482ef3514fd extends Twig_Template
{
    public function __construct(Twig_Environment $env)
    {
        parent::__construct($env);

        // line 1
        $this->parent = $this->loadTemplate("layout/layout.twig", "search.twig", 1);
        $this->blocks = array(
            'title' => array($this, 'block_title'),
            'body_class' => array($this, 'block_body_class'),
            'page_content' => array($this, 'block_page_content'),
            'js_search' => array($this, 'block_js_search'),
        );
    }

    protected function doGetParent(array $context)
    {
        return "layout/layout.twig";
    }

    protected function doDisplay(array $context, array $blocks = array())
    {
        // line 2
        $context["__internal_4d5ba30a225281f2a814929c3fda37109b35b5b3792760da86991b110c2f465a"] = $this->loadTemplate("macros.twig", "search.twig", 2);
        // line 1
        $this->parent->display($context, array_merge($this->blocks, $blocks));
    }

    // line 3
    public function block_title($context, array $blocks = array())
    {
        echo "Search | ";
        $this->displayParentBlock("title", $context, $blocks);
    }

    // line 4
    public function block_body_class($context, array $blocks = array())
    {
        echo "search-page";
    }

    // line 6
    public function block_page_content($context, array $blocks = array())
    {
        // line 7
        echo "
    <div class=\"page-header\">
        <h1>Search</h1>
    </div>

    <p>This page allows you to search through the API documentation for
    specific terms. Enter your search words into the box below and click
    \"submit\". The search will be performed on namespaces, clases, interfaces,
    traits, functions, and methods.</p>

    <form class=\"form-inline\" role=\"form\" action=\"";
        // line 17
        echo twig_escape_filter($this->env, $this->env->getExtension('sami')->pathForStaticFile($context, "search.html"), "html", null, true);
        echo "\" method=\"GET\">
        <div class=\"form-group\">
            <label class=\"sr-only\" for=\"search\">Search</label>
            <input type=\"search\" class=\"form-control\" name=\"search\" id=\"search\" placeholder=\"Search\">
        </div>
        <button type=\"submit\" class=\"btn btn-default\">submit</button>
    </form>

    <h2>Search Results</h2>

    <div class=\"container-fluid\">
        <ul class=\"search-results\"></ul>
    </div>

    ";
        // line 31
        $this->displayBlock("js_search", $context, $blocks);
        echo "

";
    }

    // line 35
    public function block_js_search($context, array $blocks = array())
    {
        // line 36
        echo "    <script type=\"text/javascript\">

        (function() {
            var term = Sami.cleanSearchTerm();

            if (!term) {
                \$('h2').hide();
                return;
            }

            \$('#search').val(term);
            var container = \$('.search-results');

            // create the search index
            ";
        // line 50
        $this->displayBlock("search_index", $context, $blocks);
        echo "

            var results = Sami.search(term);
            var len = results.length;

            if (len == 0) {
                container.html('No results were found');
                return;
            }

            for (var i = 0, text_content = ''; i < len; i++) {

                var ele = results[i];
                var contents = '<li><h2 class=\"clearfix\">';
                var class_name = Sami.getSearchClass(ele.type);
                contents += '<a href=\"' + ele.link + '\">' + ele.name + '</a>';
                contents += '<div class=\"search-type type-' + ele.type + '\"><span class=\"pull-right label ' + class_name + '\">' + ele.type + '</span></div>';
                contents += '</h2>';

                if (ele.fromName && ele.fromLink) {
                    contents += '<div class=\"search-from\">from <a href=\"' + ele.fromLink + '\">' + ele.fromName + '</a></div>';
                }

                contents += '<div class=\"search-description\">';

                // Add description, decode entities, and strip wrapping quotes
                if (ele.doc) {
                    text_content = \$('<p>' + ele.doc + '</p>').text();
                    if (text_content[0] == '\"') {
                        text_content = text_content.substring(1);
                    }
                    if (text_content[text_content.length - 1] == '\"') {
                        text_content = text_content.substring(0, text_content.length - 1);
                    }
                    contents += text_content;
                }

                contents += '</div></li>';
                container.append(\$(contents));
            }
        })();
    </script>
";
    }

    public function getTemplateName()
    {
        return "search.twig";
    }

    public function isTraitable()
    {
        return false;
    }

    public function getDebugInfo()
    {
        return array (  105 => 50,  89 => 36,  86 => 35,  79 => 31,  62 => 17,  50 => 7,  47 => 6,  41 => 4,  34 => 3,  30 => 1,  28 => 2,  11 => 1,);
    }
}
/* {% extends "layout/layout.twig" %}*/
/* {% from "macros.twig" import class_link, namespace_link, method_link, property_link %}*/
/* {% block title %}Search | {{ parent() }}{% endblock %}*/
/* {% block body_class 'search-page' %}*/
/* */
/* {% block page_content %}*/
/* */
/*     <div class="page-header">*/
/*         <h1>Search</h1>*/
/*     </div>*/
/* */
/*     <p>This page allows you to search through the API documentation for*/
/*     specific terms. Enter your search words into the box below and click*/
/*     "submit". The search will be performed on namespaces, clases, interfaces,*/
/*     traits, functions, and methods.</p>*/
/* */
/*     <form class="form-inline" role="form" action="{{ path('search.html') }}" method="GET">*/
/*         <div class="form-group">*/
/*             <label class="sr-only" for="search">Search</label>*/
/*             <input type="search" class="form-control" name="search" id="search" placeholder="Search">*/
/*         </div>*/
/*         <button type="submit" class="btn btn-default">submit</button>*/
/*     </form>*/
/* */
/*     <h2>Search Results</h2>*/
/* */
/*     <div class="container-fluid">*/
/*         <ul class="search-results"></ul>*/
/*     </div>*/
/* */
/*     {{ block('js_search') }}*/
/* */
/* {% endblock %}*/
/* */
/* {% block js_search %}*/
/*     <script type="text/javascript">*/
/* */
/*         (function() {*/
/*             var term = Sami.cleanSearchTerm();*/
/* */
/*             if (!term) {*/
/*                 $('h2').hide();*/
/*                 return;*/
/*             }*/
/* */
/*             $('#search').val(term);*/
/*             var container = $('.search-results');*/
/* */
/*             // create the search index*/
/*             {{ block('search_index') }}*/
/* */
/*             var results = Sami.search(term);*/
/*             var len = results.length;*/
/* */
/*             if (len == 0) {*/
/*                 container.html('No results were found');*/
/*                 return;*/
/*             }*/
/* */
/*             for (var i = 0, text_content = ''; i < len; i++) {*/
/* */
/*                 var ele = results[i];*/
/*                 var contents = '<li><h2 class="clearfix">';*/
/*                 var class_name = Sami.getSearchClass(ele.type);*/
/*                 contents += '<a href="' + ele.link + '">' + ele.name + '</a>';*/
/*                 contents += '<div class="search-type type-' + ele.type + '"><span class="pull-right label ' + class_name + '">' + ele.type + '</span></div>';*/
/*                 contents += '</h2>';*/
/* */
/*                 if (ele.fromName && ele.fromLink) {*/
/*                     contents += '<div class="search-from">from <a href="' + ele.fromLink + '">' + ele.fromName + '</a></div>';*/
/*                 }*/
/* */
/*                 contents += '<div class="search-description">';*/
/* */
/*                 // Add description, decode entities, and strip wrapping quotes*/
/*                 if (ele.doc) {*/
/*                     text_content = $('<p>' + ele.doc + '</p>').text();*/
/*                     if (text_content[0] == '"') {*/
/*                         text_content = text_content.substring(1);*/
/*                     }*/
/*                     if (text_content[text_content.length - 1] == '"') {*/
/*                         text_content = text_content.substring(0, text_content.length - 1);*/
/*                     }*/
/*                     contents += text_content;*/
/*                 }*/
/* */
/*                 contents += '</div></li>';*/
/*                 container.append($(contents));*/
/*             }*/
/*         })();*/
/*     </script>*/
/* {% endblock %}*/
/* */


cache/master/twig/10/10abef489b700f70470b0dc7a784ec00779ced43d6106a45eb3416ce77ba65d5.php
<?php

/* traits.twig */
class __TwigTemplate_c03ef46fcf749457d0f7e4f052edc367c5436d67590a69f772c08dafd041d265 extends Twig_Template
{
    public function __construct(Twig_Environment $env)
    {
        parent::__construct($env);

        // line 1
        $this->parent = $this->loadTemplate("layout/layout.twig", "traits.twig", 1);
        $this->blocks = array(
            'title' => array($this, 'block_title'),
            'body_class' => array($this, 'block_body_class'),
            'page_content' => array($this, 'block_page_content'),
        );
    }

    protected function doGetParent(array $context)
    {
        return "layout/layout.twig";
    }

    protected function doDisplay(array $context, array $blocks = array())
    {
        // line 2
        $context["__internal_08063baed92e4826268e135ee029cd0a3ad929ef17d4562b09330b53fdd24c24"] = $this->loadTemplate("macros.twig", "traits.twig", 2);
        // line 1
        $this->parent->display($context, array_merge($this->blocks, $blocks));
    }

    // line 3
    public function block_title($context, array $blocks = array())
    {
        echo "Traits | ";
        $this->displayParentBlock("title", $context, $blocks);
    }

    // line 4
    public function block_body_class($context, array $blocks = array())
    {
        echo "traits";
    }

    // line 6
    public function block_page_content($context, array $blocks = array())
    {
        // line 7
        echo "    <div class=\"page-header\">
        <h1>Traits</h1>
    </div>

    <div class=\"container-fluid underlined\">
        ";
        // line 12
        $context['_parent'] = $context;
        $context['_seq'] = twig_ensure_traversable((isset($context["classes"]) ? $context["classes"] : $this->getContext($context, "classes")));
        foreach ($context['_seq'] as $context["_key"] => $context["class"]) {
            // line 13
            echo "            ";
            if ($this->getAttribute($context["class"], "trait", array())) {
                // line 14
                echo "                <div class=\"row\">
                    <div class=\"col-md-6\">
                        ";
                // line 16
                echo $context["__internal_08063baed92e4826268e135ee029cd0a3ad929ef17d4562b09330b53fdd24c24"]->getclass_link($context["class"], true);
                echo "
                    </div>
                    <div class=\"col-md-6\">
                        ";
                // line 19
                echo $this->env->getExtension('sami')->parseDesc($context, $this->getAttribute($context["class"], "shortdesc", array()), $context["class"]);
                echo "
                    </div>
                </div>
            ";
            }
            // line 23
            echo "        ";
        }
        $_parent = $context['_parent'];
        unset($context['_seq'], $context['_iterated'], $context['_key'], $context['class'], $context['_parent'], $context['loop']);
        $context = array_intersect_key($context, $_parent) + $_parent;
        // line 24
        echo "    </div>
";
    }

    public function getTemplateName()
    {
        return "traits.twig";
    }

    public function isTraitable()
    {
        return false;
    }

    public function getDebugInfo()
    {
        return array (  86 => 24,  80 => 23,  73 => 19,  67 => 16,  63 => 14,  60 => 13,  56 => 12,  49 => 7,  46 => 6,  40 => 4,  33 => 3,  29 => 1,  27 => 2,  11 => 1,);
    }
}
/* {% extends "layout/layout.twig" %}*/
/* {% from "macros.twig" import class_link %}*/
/* {% block title %}Traits | {{ parent() }}{% endblock %}*/
/* {% block body_class 'traits' %}*/
/* */
/* {% block page_content %}*/
/*     <div class="page-header">*/
/*         <h1>Traits</h1>*/
/*     </div>*/
/* */
/*     <div class="container-fluid underlined">*/
/*         {% for class in classes %}*/
/*             {% if class.trait %}*/
/*                 <div class="row">*/
/*                     <div class="col-md-6">*/
/*                         {{ class_link(class, true) }}*/
/*                     </div>*/
/*                     <div class="col-md-6">*/
/*                         {{ class.shortdesc|desc(class) }}*/
/*                     </div>*/
/*                 </div>*/
/*             {% endif %}*/
/*         {% endfor %}*/
/*     </div>*/
/* {% endblock %}*/
/* */


cache/master/twig/5f/5f8c5b78dfdddb7bae1f394433f7a73361dfa2674b9825235410a356d14313d9.php
<?php

/* interfaces.twig */
class __TwigTemplate_a670aa618829b01e809a02f9a1b29fbc4802e81efa1c9c9e0f4f599f821a94a1 extends Twig_Template
{
    public function __construct(Twig_Environment $env)
    {
        parent::__construct($env);

        // line 1
        $this->parent = $this->loadTemplate("layout/layout.twig", "interfaces.twig", 1);
        $this->blocks = array(
            'title' => array($this, 'block_title'),
            'body_class' => array($this, 'block_body_class'),
            'page_content' => array($this, 'block_page_content'),
        );
    }

    protected function doGetParent(array $context)
    {
        return "layout/layout.twig";
    }

    protected function doDisplay(array $context, array $blocks = array())
    {
        // line 2
        $context["__internal_9fd3c9a87638bcb94d21d79116a84c9d7f8d3310b0aa76ce0f481ef6153dd337"] = $this->loadTemplate("macros.twig", "interfaces.twig", 2);
        // line 1
        $this->parent->display($context, array_merge($this->blocks, $blocks));
    }

    // line 3
    public function block_title($context, array $blocks = array())
    {
        echo "Interfaces | ";
        $this->displayParentBlock("title", $context, $blocks);
    }

    // line 4
    public function block_body_class($context, array $blocks = array())
    {
        echo "interfaces";
    }

    // line 6
    public function block_page_content($context, array $blocks = array())
    {
        // line 7
        echo "    <div class=\"page-header\">
        <h1>Interfaces</h1>
    </div>

    ";
        // line 11
        echo $context["__internal_9fd3c9a87638bcb94d21d79116a84c9d7f8d3310b0aa76ce0f481ef6153dd337"]->getrender_classes((isset($context["interfaces"]) ? $context["interfaces"] : $this->getContext($context, "interfaces")));
        echo "
";
    }

    public function getTemplateName()
    {
        return "interfaces.twig";
    }

    public function isTraitable()
    {
        return false;
    }

    public function getDebugInfo()
    {
        return array (  55 => 11,  49 => 7,  46 => 6,  40 => 4,  33 => 3,  29 => 1,  27 => 2,  11 => 1,);
    }
}
/* {% extends "layout/layout.twig" %}*/
/* {% from "macros.twig" import render_classes %}*/
/* {% block title %}Interfaces | {{ parent() }}{% endblock %}*/
/* {% block body_class 'interfaces' %}*/
/* */
/* {% block page_content %}*/
/*     <div class="page-header">*/
/*         <h1>Interfaces</h1>*/
/*     </div>*/
/* */
/*     {{ render_classes(interfaces) }}*/
/* {% endblock %}*/
/* */


cache/master/twig/d3/d306d845cf6a8ef83bfd6e2d7f89a0010ae21908218de53ef59e7f258d46c1f1.php
<?php

/* classes.twig */
class __TwigTemplate_9a678aef793fca7e4fb479d8aceced02dace3aa9a52ddbb6e08b7ac9147f6ee9 extends Twig_Template
{
    public function __construct(Twig_Environment $env)
    {
        parent::__construct($env);

        // line 1
        $this->parent = $this->loadTemplate("layout/layout.twig", "classes.twig", 1);
        $this->blocks = array(
            'title' => array($this, 'block_title'),
            'body_class' => array($this, 'block_body_class'),
            'page_content' => array($this, 'block_page_content'),
        );
    }

    protected function doGetParent(array $context)
    {
        return "layout/layout.twig";
    }

    protected function doDisplay(array $context, array $blocks = array())
    {
        // line 2
        $context["__internal_89a510c5157934aa58fb5e835857365e06038629ff84a31e4e23f8187c70a8fc"] = $this->loadTemplate("macros.twig", "classes.twig", 2);
        // line 1
        $this->parent->display($context, array_merge($this->blocks, $blocks));
    }

    // line 3
    public function block_title($context, array $blocks = array())
    {
        echo "All Classes | ";
        $this->displayParentBlock("title", $context, $blocks);
    }

    // line 4
    public function block_body_class($context, array $blocks = array())
    {
        echo "classes";
    }

    // line 6
    public function block_page_content($context, array $blocks = array())
    {
        // line 7
        echo "    <div class=\"page-header\">
        <h1>Classes</h1>
    </div>

    ";
        // line 11
        echo $context["__internal_89a510c5157934aa58fb5e835857365e06038629ff84a31e4e23f8187c70a8fc"]->getrender_classes((isset($context["classes"]) ? $context["classes"] : $this->getContext($context, "classes")));
        echo "
";
    }

    public function getTemplateName()
    {
        return "classes.twig";
    }

    public function isTraitable()
    {
        return false;
    }

    public function getDebugInfo()
    {
        return array (  55 => 11,  49 => 7,  46 => 6,  40 => 4,  33 => 3,  29 => 1,  27 => 2,  11 => 1,);
    }
}
/* {% extends "layout/layout.twig" %}*/
/* {% from "macros.twig" import render_classes %}*/
/* {% block title %}All Classes | {{ parent() }}{% endblock %}*/
/* {% block body_class 'classes' %}*/
/* */
/* {% block page_content %}*/
/*     <div class="page-header">*/
/*         <h1>Classes</h1>*/
/*     </div>*/
/* */
/*     {{ render_classes(classes) }}*/
/* {% endblock %}*/
/* */


cache/master/twig/ba/baa18224487ff5f619ea618aa1aa65ae9572ebeee186df0a4cd93bbcad7e1d33.php
<?php

/* doc-index.twig */
class __TwigTemplate_b7ad7d5daa07463ced854e264e048e136f494ff640d763662586fb9a0f17816c extends Twig_Template
{
    public function __construct(Twig_Environment $env)
    {
        parent::__construct($env);

        // line 1
        $this->parent = $this->loadTemplate("layout/layout.twig", "doc-index.twig", 1);
        $this->blocks = array(
            'title' => array($this, 'block_title'),
            'body_class' => array($this, 'block_body_class'),
            'page_content' => array($this, 'block_page_content'),
        );
    }

    protected function doGetParent(array $context)
    {
        return "layout/layout.twig";
    }

    protected function doDisplay(array $context, array $blocks = array())
    {
        // line 2
        $context["__internal_1c386eb7ef5c9e1e3b0592558e1cabd3813a0f2c330bb1fa2f3976ba3dfb7202"] = $this->loadTemplate("macros.twig", "doc-index.twig", 2);
        // line 1
        $this->parent->display($context, array_merge($this->blocks, $blocks));
    }

    // line 3
    public function block_title($context, array $blocks = array())
    {
        echo "Index | ";
        $this->displayParentBlock("title", $context, $blocks);
    }

    // line 4
    public function block_body_class($context, array $blocks = array())
    {
        echo "doc-index";
    }

    // line 6
    public function block_page_content($context, array $blocks = array())
    {
        // line 7
        echo "
    <div class=\"page-header\">
        <h1>Index</h1>
    </div>

    <ul class=\"pagination\">
        ";
        // line 13
        $context['_parent'] = $context;
        $context['_seq'] = twig_ensure_traversable(range("A", "Z"));
        foreach ($context['_seq'] as $context["_key"] => $context["letter"]) {
            // line 14
            echo "            ";
            if (($this->getAttribute((isset($context["items"]) ? $context["items"] : null), $context["letter"], array(), "array", true, true) && (twig_length_filter($this->env, $this->getAttribute((isset($context["items"]) ? $context["items"] : $this->getContext($context, "items")), $context["letter"], array(), "array")) > 1))) {
                // line 15
                echo "                <li><a href=\"#letter";
                echo $context["letter"];
                echo "\">";
                echo $context["letter"];
                echo "</a></li>
            ";
            } else {
                // line 17
                echo "                <li class=\"disabled\"><a href=\"#letter";
                echo $context["letter"];
                echo "\">";
                echo $context["letter"];
                echo "</a></li>
            ";
            }
            // line 19
            echo "        ";
        }
        $_parent = $context['_parent'];
        unset($context['_seq'], $context['_iterated'], $context['_key'], $context['letter'], $context['_parent'], $context['loop']);
        $context = array_intersect_key($context, $_parent) + $_parent;
        // line 20
        echo "    </ul>

    ";
        // line 22
        $context['_parent'] = $context;
        $context['_seq'] = twig_ensure_traversable((isset($context["items"]) ? $context["items"] : $this->getContext($context, "items")));
        foreach ($context['_seq'] as $context["letter"] => $context["elements"]) {
            // line 23
            echo "<h2 id=\"letter";
            echo $context["letter"];
            echo "\">";
            echo $context["letter"];
            echo "</h2>
        <dl id=\"index\">";
            // line 25
            $context['_parent'] = $context;
            $context['_seq'] = twig_ensure_traversable($context["elements"]);
            foreach ($context['_seq'] as $context["_key"] => $context["element"]) {
                // line 26
                $context["type"] = $this->getAttribute($context["element"], 0, array(), "array");
                // line 27
                $context["value"] = $this->getAttribute($context["element"], 1, array(), "array");
                // line 28
                if (("class" == (isset($context["type"]) ? $context["type"] : $this->getContext($context, "type")))) {
                    // line 29
                    echo "<dt>";
                    echo $context["__internal_1c386eb7ef5c9e1e3b0592558e1cabd3813a0f2c330bb1fa2f3976ba3dfb7202"]->getclass_link((isset($context["value"]) ? $context["value"] : $this->getContext($context, "value")));
                    if ((isset($context["has_namespaces"]) ? $context["has_namespaces"] : $this->getContext($context, "has_namespaces"))) {
                        echo " &mdash; <em>Class in namespace ";
                        echo $context["__internal_1c386eb7ef5c9e1e3b0592558e1cabd3813a0f2c330bb1fa2f3976ba3dfb7202"]->getnamespace_link($this->getAttribute((isset($context["value"]) ? $context["value"] : $this->getContext($context, "value")), "namespace", array()));
                    }
                    echo "</em></dt>
                    <dd>";
                    // line 30
                    echo $this->env->getExtension('sami')->parseDesc($context, $this->getAttribute((isset($context["value"]) ? $context["value"] : $this->getContext($context, "value")), "shortdesc", array()), (isset($context["value"]) ? $context["value"] : $this->getContext($context, "value")));
                    echo "</dd>";
                } elseif (("method" ==                 // line 31
(isset($context["type"]) ? $context["type"] : $this->getContext($context, "type")))) {
                    // line 32
                    echo "<dt>";
                    echo $context["__internal_1c386eb7ef5c9e1e3b0592558e1cabd3813a0f2c330bb1fa2f3976ba3dfb7202"]->getmethod_link((isset($context["value"]) ? $context["value"] : $this->getContext($context, "value")));
                    echo "() &mdash; <em>Method in class ";
                    echo $context["__internal_1c386eb7ef5c9e1e3b0592558e1cabd3813a0f2c330bb1fa2f3976ba3dfb7202"]->getclass_link($this->getAttribute((isset($context["value"]) ? $context["value"] : $this->getContext($context, "value")), "class", array()));
                    echo "</em></dt>
                    <dd>";
                    // line 33
                    echo $this->env->getExtension('sami')->parseDesc($context, $this->getAttribute((isset($context["value"]) ? $context["value"] : $this->getContext($context, "value")), "shortdesc", array()), $this->getAttribute((isset($context["value"]) ? $context["value"] : $this->getContext($context, "value")), "class", array()));
                    echo "</dd>";
                } elseif (("property" ==                 // line 34
(isset($context["type"]) ? $context["type"] : $this->getContext($context, "type")))) {
                    // line 35
                    echo "<dt>\$";
                    echo $context["__internal_1c386eb7ef5c9e1e3b0592558e1cabd3813a0f2c330bb1fa2f3976ba3dfb7202"]->getproperty_link((isset($context["value"]) ? $context["value"] : $this->getContext($context, "value")));
                    echo " &mdash; <em>Property in class ";
                    echo $context["__internal_1c386eb7ef5c9e1e3b0592558e1cabd3813a0f2c330bb1fa2f3976ba3dfb7202"]->getclass_link($this->getAttribute((isset($context["value"]) ? $context["value"] : $this->getContext($context, "value")), "class", array()));
                    echo "</em></dt>
                    <dd>";
                    // line 36
                    echo $this->env->getExtension('sami')->parseDesc($context, $this->getAttribute((isset($context["value"]) ? $context["value"] : $this->getContext($context, "value")), "shortdesc", array()), $this->getAttribute((isset($context["value"]) ? $context["value"] : $this->getContext($context, "value")), "class", array()));
                    echo "</dd>";
                }
            }
            $_parent = $context['_parent'];
            unset($context['_seq'], $context['_iterated'], $context['_key'], $context['element'], $context['_parent'], $context['loop']);
            $context = array_intersect_key($context, $_parent) + $_parent;
            // line 39
            echo "        </dl>";
        }
        $_parent = $context['_parent'];
        unset($context['_seq'], $context['_iterated'], $context['letter'], $context['elements'], $context['_parent'], $context['loop']);
        $context = array_intersect_key($context, $_parent) + $_parent;
    }

    public function getTemplateName()
    {
        return "doc-index.twig";
    }

    public function isTraitable()
    {
        return false;
    }

    public function getDebugInfo()
    {
        return array (  152 => 39,  144 => 36,  137 => 35,  135 => 34,  132 => 33,  125 => 32,  123 => 31,  120 => 30,  111 => 29,  109 => 28,  107 => 27,  105 => 26,  101 => 25,  94 => 23,  90 => 22,  86 => 20,  80 => 19,  72 => 17,  64 => 15,  61 => 14,  57 => 13,  49 => 7,  46 => 6,  40 => 4,  33 => 3,  29 => 1,  27 => 2,  11 => 1,);
    }
}
/* {% extends "layout/layout.twig" %}*/
/* {% from "macros.twig" import class_link, namespace_link, method_link, property_link %}*/
/* {% block title %}Index | {{ parent() }}{% endblock %}*/
/* {% block body_class 'doc-index' %}*/
/* */
/* {% block page_content %}*/
/* */
/*     <div class="page-header">*/
/*         <h1>Index</h1>*/
/*     </div>*/
/* */
/*     <ul class="pagination">*/
/*         {% for letter in 'A'..'Z' %}*/
/*             {% if items[letter] is defined and items[letter]|length > 1 %}*/
/*                 <li><a href="#letter{{ letter|raw }}">{{ letter|raw }}</a></li>*/
/*             {% else %}*/
/*                 <li class="disabled"><a href="#letter{{ letter|raw }}">{{ letter|raw }}</a></li>*/
/*             {% endif %}*/
/*         {% endfor %}*/
/*     </ul>*/
/* */
/*     {% for letter, elements in items -%}*/
/*         <h2 id="letter{{ letter|raw }}">{{ letter|raw }}</h2>*/
/*         <dl id="index">*/
/*             {%- for element in elements %}*/
/*                 {%- set type = element[0] %}*/
/*                 {%- set value = element[1] %}*/
/*                 {%- if 'class' == type -%}*/
/*                     <dt>{{ class_link(value) }}{% if has_namespaces %} &mdash; <em>Class in namespace {{ namespace_link(value.namespace) }}{% endif %}</em></dt>*/
/*                     <dd>{{ value.shortdesc|desc(value) }}</dd>*/
/*                 {%- elseif 'method' == type -%}*/
/*                     <dt>{{ method_link(value) }}() &mdash; <em>Method in class {{ class_link(value.class) }}</em></dt>*/
/*                     <dd>{{ value.shortdesc|desc(value.class) }}</dd>*/
/*                 {%- elseif 'property' == type -%}*/
/*                     <dt>${{ property_link(value) }} &mdash; <em>Property in class {{ class_link(value.class) }}</em></dt>*/
/*                     <dd>{{ value.shortdesc|desc(value.class) }}</dd>*/
/*                 {%- endif %}*/
/*             {%- endfor %}*/
/*         </dl>*/
/*     {%- endfor %}*/
/* {% endblock %}*/
/* */
